librisworks/src/main/groovy/se/kb/libris/mergeworks/compare/Classification.groovy

```diff
@@ -79,7 +73,94 @@ class Classification extends StuffSet {
         Integer.parseInt((edition ?: "0").replaceAll("[^0-9]", ""))
     }
 
-    boolean n(a, b) {
-        norm[a]?.any { it == b || n(it, b) }
+    static String normalizeSabCode(String sab) {
+        sab.replaceFirst(~/^h/, 'H').with {
+            it =~ /bf:|z/ ? it : it.replaceAll(~/\s+/, '')
+        }
+    }
+
+    static String tryMergeSabCodes(String a, String b) {
+        if (a == b) {
+            return a
+        }
+        if (sabPrecedes(a, b)) {
+            return a
+        }
+        if (sabPrecedes(b, a)) {
+            return b
+        }
+        return null
+    }
+
+    static sabPrecedes(String a, String b) {
+        def (equal, startsWith) = sabPrecedenceRules
+        def startsWithExceptions = ~/^Hcb|^Hdab/
+        def preferred = equal[b] ?: startsWith.find { b.startsWith(it.key) }?.value
+        if (preferred && !(a =~ startsWithExceptions)) {
```

Comment:
`startsWithExceptions` also apply to "equals?"
In that case the naming is a bit confusing.